network/network.go

```diff
@@ -74,6 +75,32 @@ type Adapter interface {
 	UnRegisterChannel(channel channels.Channel) error
 }
 
+// Middleware is one of the networking layer interfaces in Flow (i.e., Network, Adapter, and Middleware). It represents the interface that networking layer
+// offers to lower level networking components such as libp2p. It is responsible for subscribing to and unsubscribing
+// from channels, as well as updating the addresses of all the authorized participants in the Flow protocol.
+type Middleware interface {
```

Comment:
```suggestion
type NetworkLower interface {
```

Reply:
Per our conversation, renamed `Middleware` to `Underlay`: https://github.com/onflow/flow-go/pull/4664/commits/6c01ff5c904a67607709686a972284ec0e3e558c